### PR TITLE
Update/lammps package: new variants for refactoring of code modules

### PR DESCRIPTION
**Pull request by marcodelapierre**
In this PR, I am editing the package recipe for LAMMPS.

Key modifications:
1. updated variants to take into account large refactoring of LAMMPS modules at versions `20210702` and `20210728`
2. added variants support for older modules: `user-molfile`/`molfile`, `plugin`, `user-brownian`/`brownian`
3. added support for `cray-fftw` for the `kokkos` variant
4. added shasum for bugfix version `20210929.3`
5. fixed a bug for the `mliap`/`ml-iap` variants, when used together with the `python` variant

Modification number 1. took some time, and I did my best to test it across a small set of builds.
The refactoring involved renaming of a number of LAMMPS modules (aka packages within LAMMPS); I updated the recipe accordingly, by 
* adding new variants
* making sure previous functionalities also work with new variants
* adding version conflicts for old/new variants
As a result, the size of the recipe grew quite a bit, especially due to these latter version conflicts (about 200 lines).

Happy to talk more about this strategy.

### CODE REVIEW

**Comment by tldahlgren** [comment]
@adamjstewart  @junghans @amd-toolchain-support @sethrj @tgamblin As significant contributors to this package over the  last year, would any of you be interested in commenting on this PR?

**Comment by tldahlgren** [comment]
@adamjstewart  @junghans @amd-toolchain-support @sethrj @tgamblin As significant contributors to this package over the  last year, would any of you be interested in commenting on this PR?

**Comment by alalazo** [comment]
@spackbot maintainers

**Comment by marcodelapierre** [comment]
Any updates on this space? Thanks

**Comment by tldahlgren** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/lammps/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/lammps/package.py:191]: Did you intend to get rid of the `+user-misc` conflict?

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.  Seems reasonable.  You could use for-loops on some of the conflicts to tighten the code but that's only a suggestion.

**Comment by marcodelapierre** [comment]
Thanks for the approval @tldahlgren .
Not sure of the best approach to maintain the ever growing list of modular packages and conflicts in a readable way.
I am tempted not to further edit the current formatting of the conflicts in this PR, and leave it to future PRs instead.

**Pull request merged**
